src/containers/CreateBroadcastableTransaction/CreateBroadcastableTransactionIndex.tsx

```diff
@@ -0,0 +1,155 @@
+import * as Yup from 'yup';
+import { Button } from '~/components';
+import type {
+  BroadcastableSweepTransaction,
+  createAdaBroadcastableSweepTransactionParameters,
+  createDotBroadcastableSweepTransactionParameters,
+  createSolBroadcastableSweepTransactionParameters,
+} from '~/utils/types';
+import { assert, safeEnv } from '~/helpers';
+import { useAlertBanner } from '~/contexts';
+
+function isBroadcastableTransaction(
+  json: unknown
+): json is BroadcastableSweepTransaction {
+  const broadcastableTransaction = json as BroadcastableSweepTransaction;
+  return (
+    broadcastableTransaction &&
+    broadcastableTransaction.length !== undefined &&
+    broadcastableTransaction.length > 0 &&
+    broadcastableTransaction[0].serializedTx !== undefined
+  );
+}
+
+function isSignedTransaction(
+  json: unknown
+): json is
+  | createAdaBroadcastableSweepTransactionParameters
+  | createDotBroadcastableSweepTransactionParameters
+  | createSolBroadcastableSweepTransactionParameters {
+  const signedTransaction = json as
+    | createAdaBroadcastableSweepTransactionParameters
+    | createDotBroadcastableSweepTransactionParameters
+    | createSolBroadcastableSweepTransactionParameters;
+  return (
+    signedTransaction &&
+    signedTransaction.signatureShares !== undefined &&
+    signedTransaction.signatureShares.length > 0
+  );
+}
+
+export function CreateBroadcastableTransactionIndex() {
+  const { env } = useParams<{ env: string }>();
+  const [, setAlert] = useAlertBanner();
+
+  const environment = safeEnv(env);
+  const navigate = useNavigate();
+  const formik = useFormik<{ tx?: File }>({
+    initialValues: {
+      tx: undefined,
+    },
+    onSubmit: async (values, formikHelpers) => {
+      if (values.tx) {
+        formikHelpers.setSubmitting(true);
+        await window.commands.setBitGoEnvironment(environment);
+        const fileReader = new FileReader();
+        fileReader.readAsText(values.tx, 'UTF-8');
+        fileReader.onload = async event => {
+          try {
+            const tx = JSON.parse(event.target?.result as string) as
+              | createAdaBroadcastableSweepTransactionParameters
+              | createDotBroadcastableSweepTransactionParameters
+              | createSolBroadcastableSweepTransactionParameters;
+
+            assert(isSignedTransaction(tx), 'Signed transaction not found');
+
+            const coin = tx.signatureShares[0].txRequest.walletCoin;
+            const chainData = await window.queries.getChain(coin);
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            const broadcastTx: Error | BroadcastableSweepTransaction =
+              await window.commands.createBroadcastableSweepTransaction(
+                coin,
+                tx
+              );
+
+            assert(
+              isBroadcastableTransaction(broadcastTx),
+              'Broadcastable transaction not found'
+            );
```

Comment:
Shouldn't this be covered via a returned `error`?

Reply:
If the returned `broadcastTx` is not a broadcastable tx, it sounds like it should have been an error instead.